textx/scoping/providers.py

```diff
@@ -201,20 +218,20 @@ def _find_referenced_obj(p, name, cls):
             Returns:
                 None or the found object
             """
-            ret = _find_obj_fqn(p, name, cls)
+            ret = _find_obj_fqn(p, name, cls, current_obj)
             if ret:
                 return ret
             while hasattr(p, "parent"):
                 p = p.parent
-                ret = _find_obj_fqn(p, name, cls)
+                ret = _find_obj_fqn(p, name, cls, current_obj)
                 if ret:
                     return ret
             return None
 
         from textx.model import ObjCrossRef
         assert type(obj_ref) is ObjCrossRef, type(obj_ref)
         obj_cls, obj_name = obj_ref.cls, obj_ref.obj_name
-        ret = _find_referenced_obj(obj, obj_name, obj_cls)
+        ret = _find_referenced_obj(obj, obj_name, obj_cls, obj)
         if ret:
             return ret
         else:
```

Comment:
Can we just say here `return _find_referenced_obj(obj, obj_name, obj_cls)` ? 
No need for `if` afterwards I would say.

Reply:
Ok. See last comment (this is the second return None which I removed).